src/debugging-support-in-rustc.md

```diff
@@ -0,0 +1,259 @@
+enum into the DWARF and LLDB reads the DWARF to understand where is the tag field or 
+is there a tag field or is the tag slot shared with non-zero optimization etc. 
+In other words, it has enum support as well.
+
+#### Developer notes
+
+* None of the LLDB work is upstream. This [rust-lang/lldb wiki page] explains a few details. 
+* The reason for forking LLDB is that LLDB recently removed all the other language plugins 
+due to lack of maintenance.
+* LLDB has a plugin architecture but that does not work for language support. 
+* This is built on macOS because it was easier to solve. 
+* Tom has a possible solution which can enable us to ship LLDB everywhere. 
+* GDB generally works better on Linux.
+
+## DWARF and Rustc
+
+[DWARF] is the standard way compilers generate debugging information that debuggers read. 
+It is _the_ debugging format on macOS and Linux. It is a multi-language, extensible format 
+and is mostly good enough for Rust's purposes. Hence, the current implementation reuses DWARF's 
+concepts. This is true even if some of the concepts in DWARF do not align with Rust
+semantically because generally there can be some kind of mapping between the two.
+
+We have some DWARF extensions that the Rust compiler emits and the debuggers understand that 
+are _not_ in the DWARF standard.
+
+* vtable thing
+* .... Does not remember...
+[TODO: Ask Tom for this? they are documented in debuggers in comments or LLVM in comments]
+
+### Current limitations of DWARF
+
+* Traits - require a bigger change than normal to DWARF, on how to represent Traits in DWARF.
+* DWARF provides no way to differentiate between Structs and Tuples. Rust compiler emits 
+fields with `__0` and debuggers look for a sequence of such names to overcome this limitation. 
+For example, in this case the debugger would look at a field via `x.__0` instead of `x.0`. 
+This is resolved via the Rust parser in the debugger so now you can do `x.0`.
+
+DWARF relies on debuggers to know some information about platform ABI. 
+Rust does not do that all the time.
+
+
+## Developer notes
+
+This section is from the talk about certain aspects of development.
+
+### What is missing
+
+#### Ship GDB in Rustup
+
+PR and issue: https://github.com/rust-lang/rust/issues/34457
+
+Shipping GDB requires change to Rustup delivery system, so that GDB could be built as its own 
+entity so that the artifacts that are created that way can be stuck into rustup without requiring 
+it to be part of one giant Rust build. However, if we can ship GDB with rustup, it will simplify
+the development process by having compiler emit new debug info.
+
+Main issue is setting up dependencies. One such dependency is Python. That is why we have our 
+own fork of GDB because one of the drivers is patched on Rust's side to check the correct 
+version of Python (Python 2.7 in this case).
+
+[TODO: Tom relies of commong denominator - Python 2.7. But its end-of-life is near. confirm]
```

Comment:
In the video, @tromey says that the common denominator used here was Python 2.7. Given that Python 2.7 is going to retire by the end of this year (https://pythonclock.org/), maybe we should think about moving to 3.x?

Reply:
The main driver for choosing 2.7 for Linux builds -- a project I never finished -- was that it's difficult otherwise to build a version that will work everywhere.  Unfortunately the Python stable ABI (https://docs.python.org/3/c-api/stable.html) doesn't seem sufficient for gdb's needs.